docs/guide/models/execution.rst

```diff
@@ -0,0 +1,29 @@
+Protocols and the Execution Model
+=================================
+
+Protocols in OpenFE are built on a flexible execution model. 
+Result objects are shaped by this model, and therefore some basic
+background on it can be useful when looking into the details of simulation
+results, even if most users don't need to work with the details of the
+execution model.
+
+Each protocol involves a number of steps (called ``ProtocolUnit``\ s) which occur in
+some order. Formally, this is described by a directed acyclic graph (DAG),
+so the collection of steps to run is called a ``ProtocolDAG``. A
```

Comment:
could you add a todo to add a figure of a DAG? Something like the NEQ protocol would make a good figure here (or we can make an arbitrary one as necessary).